### PR TITLE
352 newest simulation first

### LINKED ISSUE #352: Newest simulation first
When having more than 6 simulation, the oldest are displayed first. I would like to see the newest (inluding currently running) first.

Here is the default window:
image

After moving to 22 simulation per page:
image

### PR DESCRIPTION
**Pull request by grzanka**
Improved paging calculation, needs change in UI to work (see https://github.com/yaptide/ui/pull/1072).
The changes in UI are driven by new couning schema, we count from 1, not from 0.
Instead of manually calculating pageing we use SQLAlchemy features.
Few improvements in the testing framework are included here

### CODE REVIEW

File: yaptide/routes/user_routes.py [hunk 1]
```diff
@@ -17,8 +15,8 @@
 from yaptide.routes.utils.response_templates import yaptide_response, error_validation_response
 
 MAX_PAGE_SIZE = 100
-DEFAULT_PAGE_SIZE = 10
-DEFAULT_PAGE_IDX = 0
+DEFAULT_PAGE_SIZE = 6
+DEFAULT_PAGE_IDX = 1
```
Comment (Derstilon) [inline yaptide/routes/user_routes.py]: remove or use as default value when the request didn't provide any value.

File: yaptide/routes/user_routes.py [hunk 2]
```diff
@@ -53,34 +51,15 @@ def get(user: UserModel):
...
-        page_idx = (
-            params_dict['page_idx']
-            if params_dict['page_idx'] > -1 and params_dict['page_idx'] < page_count
-            else DEFAULT_PAGE_IDX
-        )
-        simulations = simulations[page_size*page_idx: min(page_size*(page_idx+1), sim_count)]
+        # Query the database for the paginated results
+        sorting = desc if params_dict['order_type'] == OrderType.DESCEND.value else asc
+        query = SimulationModel.query.filter_by(user_id=user.id).order_by(sorting(params_dict['order_by']))
+        pagination = query.paginate(page=params_dict['page_idx'], per_page=params_dict['page_size'], error_out=False)
```
Comment (grzanka) [inline yaptide/routes/user_routes.py:58]: Consider using default values if not submitted from frontend

Comment (grzanka) [reply yaptide/routes/user_routes.py:58]: Add tests for missing values

File: yaptide/routes/user_routes.py (hunk 1 above)
Comment (grzanka) [reply yaptide/routes/user_routes.py]: Done in 3ce3177

File: yaptide/routes/user_routes.py (hunk 2 above)
Comment (grzanka) [reply yaptide/routes/user_routes.py:58]: Done in last commit

**Review by Derstilon — APPROVED** [review]

**Pull request merged**